src/shared/udev-util.c

```diff
@@ -108,8 +110,31 @@ int udev_parse_config_full(
         return 0;
 }
 
+static int device_new_from_dev_path(const char *devlink, sd_device **ret_device) {
+        struct stat st;
+        int r;
+
+        assert(devlink);
+
+        r = stat(devlink, &st);
+        if (r < 0 && errno == ENOENT)
+                return -ENOENT;
+        if (r < 0)
+                return log_error_errno(errno, "Failed to stat() %s: %m", devlink);
+        if (r == 0) {
```

Comment:
this if check can just be dropped, no, it's the only possible result here...

Reply:
true, was a leftover, removed